lib/client/ippool_e2e_test.go

```diff
@@ -265,4 +261,52 @@ var _ = Describe("IPPool tests", func() {
 			},
 		),
 	)
+
+	Describe("Checking operations perform data validation", func() {
+		testutils.CleanEtcd()
+		c, _ := testutils.NewClient("")
+		var err error
+		valErrorType := reflect.TypeOf(cerrors.ErrorValidation{})
+
+		// Step-1: Test data validation occurs on create.
+		It("should invoke validation failure", func() {
+			By("Creating a pool with small CIDR")
+			_, err = c.IPPools().Create(&api.IPPool{
+				Metadata: api.IPPoolMetadata{CIDR: testutils.MustParseCIDR("10.10.10.0/30")},
+				Spec:     api.IPPoolSpec{},
+			})
+
+			Expect(err).To(HaveOccurred())
+			Expect(reflect.TypeOf(err)).To(Equal(valErrorType))
+		})
+
+		// Step-2: Test data validation occurs on apply.
+		It("should invoke validation failure", func() {
+			By("Applying a pool with small CIDR")
+			_, err = c.IPPools().Apply(&api.IPPool{
+				Metadata: api.IPPoolMetadata{CIDR: testutils.MustParseCIDR("aa:bb::cc/125")},
+				Spec:     api.IPPoolSpec{},
+			})
+			Expect(err).To(HaveOccurred())
+			Expect(reflect.TypeOf(err)).To(Equal(valErrorType))
+		})
+
+		// Step-3: Test data validation occurs on update.
```

Comment:
Because there is no validation of IP pool metadata so it didn't seem like a very good end to end test.

I added those test in the node e2e file.